platform/ios/ios.xcodeproj/project.pbxproj

```diff
@@ -2167,7 +2167,7 @@
 				CURRENT_COMMIT_HASH = deadbeef;
 				CURRENT_SEMANTIC_VERSION = 1.0.0;
 				CURRENT_SHORT_VERSION = 1.0;
-				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
```

Comment:
Compilation speed is indeed part of the reason that the default is `dwarf` for Debug builds. I think we should keep it that way for builds within Xcode. A dSYM is unnecessary because the SDK is built with debug symbols.

`make iproj` builds a Debug build by default but strips debug symbols in a post-build step. Therefore, package.sh should be modified to override this build setting with `dwarf-with-dsym` if symbols would be stripped (see also #6531).

Please make corresponding changes for the macOS SDK. Thanks!